### PR TITLE
Add re-enabled opcodes from November 2018

### PR DESCRIPTION
**Pull request by dfoderick**
Adds the re-enabled opcodes from November 2018 fork.
OP_MUL, OP_INVERT, OP_LSHIFT, OP_RSHIFT

https://www.reddit.com/r/btc/comments/980d0s/specification_for_reenabling_old_opcodes_november/
https://docs.google.com/document/d/107FaEyiIhVTngJQbA-0U6CuWmtldiyD09gDO1jxFuAk/edit

Note the following:
* Think of this as a candidate for merging but needs more testing of edge cases.
* The bitcoind client implementation of left shift does not expand the number of bytes on the stack. In other words if one byte is being shifted then only one byte is ever put back onto the stack. Carry bits are lost. This makes some sense if you think about keeping script operations simple and "deterministic".
* I added some TODO comments to interpreter.js because it uses magic numbers. These should be gathered and placed into one section at a minimum. In future these will change as limits are lifted.

### CODE REVIEW

**Comment by ryanxcharles** [comment]
Wow, awesome work! Will review shortly.

**Comment by ryanxcharles** [comment]
Looks good to me! As mentioned on Slack - you are writing more test cases. Let us know when you are ready to merge.

File: lib/script/interpreter.js [hunk 1]
```diff
@@ -627,16 +632,21 @@ Interpreter.prototype.step = function () {
...
 
         // Disabled opcodes.
         return true
 
+      case Opcode.OP_INVERT:
+      case Opcode.OP_MUL:
+      case Opcode.OP_LSHIFT:
+      case Opcode.OP_RSHIFT:
+        // Opcodes that have been reenabled.
+        if ((self.flags & Interpreter.SCRIPT_ENABLE_MAGNETIC_OPCODES) === 0) {
```
Comment (rt121212121) [inline lib/script/interpreter.js]: Is this something that should be refactored out and not copied from the legacy BAB code?

Comment (dfoderick) [reply lib/script/interpreter.js]: You might be right. Certainly in another 6 months or 1 year it will be dead code. I will see how the rest of testing goes and I might just remove Magnetic and Monolith flags.

**Comment by ryanxcharles** [comment]
since tests pass and these are useful right now, i'm going to go ahead and merge this. if you want to add more tests there can be another PR.

**Comment by dfoderick** [comment]
Yes. There are some edge cases that were not covered. I will do another pull request.
